jetty-server/src/main/java/org/eclipse/jetty/server/handler/AllowSymLinkAliasChecker.java

```diff
@@ -99,4 +88,22 @@ public boolean check(String uri, Resource resource)
         return false;
     }
 
+    private Path unrollSymbolicLink(Path path) throws IOException
+    {
+      Path d = path.getRoot();
+      for (Path e:path)
+      {
+          d=d.resolve(e);
+
+          while (Files.exists(d) && Files.isSymbolicLink(d))
+          {
+              Path link=Files.readSymbolicLink(d);
+              if (!link.isAbsolute())
+                  link=d.resolve(link);
+              d=unrollSymbolicLink(link);
```

Comment:
It is certainly a bit of an edge case.  I have the following:
/web/logs -> /var/log -> /media/internal/logs
where /media/internal -> /media/internal-physical.

Without this extra recursion, that last symlink in the parent directory does not get resolved.